catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/MigrationTaskManager.java

```diff
@@ -45,27 +52,54 @@
 class MigrationTaskManager implements AutoCloseable {
     private static final Logger LOGGER = LoggerFactory.getLogger(MigrationTaskManager.class);
 
-    private final ListeningExecutorService taskExecutor;
-
     private final MigrationFileWriter fileWriter;
 
     private final CatalogMigratableConfig catalogConfig;
 
+    private final Supplier<ExecutorService> executorSupplier;
+
+    private ListeningExecutorService taskExecutor;
+
     private boolean failureFlag;
 
     /**
      * The creation of a task manager requires the passing of migratable configurations so that
      * task behavior can be explicitly defined (i.e. where to export, how many metacards per file,
-     * and other useful data). Also initializes the {@link MigrationFileWriter} and write directory.
+     * and other useful data). Also initializes the {@link MigrationFileWriter} and threading services.
      *
-     * @param config The set of configuration options to use.
+     * @param config Configuration object to use for customizing catalog migration process.
      */
-    public MigrationTaskManager(final CatalogMigratableConfig config,
-            final MigrationFileWriter fileWriter, final ExecutorService executor) {
-        this.taskExecutor = MoreExecutors.listeningDecorator(executor);
+    public MigrationTaskManager(@NotNull final CatalogMigratableConfig config,
+            @NotNull final MigrationFileWriter fileWriter) {
+        notNull(config, "Configuration object cannot be null");
+        notNull(fileWriter, "File writer cannot be null");
+
+        this.executorSupplier = this::createExecutorService;
```

Comment:
Really neat trick with the `Supplier` here. Like it :+1: 

I think we could simplify this a little more by either replacing the other constructor with a setter method for the `Supplier`, or have this constructor delegate to the other one:

```
public MigrationTaskManager(@NotNull final CatalogMigratableConfig config,
        @NotNull final MigrationFileWriter fileWriter) {
    this(config, fileWriter, this::createExecutorService);
}
```